packages/gatsby/src/utils/api-runner-node.js

```diff
@@ -130,10 +130,10 @@ function getLocalReporter({ activity, reporter }) {
 }
 
 function extendErrorIdWithPluginName(pluginName, errorMeta) {
-  if (typeof errorMeta === `object`) {
-    const id = errorMeta && errorMeta[`id`]
+  if (typeof errorMeta === `object` && errorMeta[`id`]) {
+    const id = errorMeta[`id`]
```

Comment:
Up to you :)

```suggestion
  const id = errorMeta?.id
  if (id) {
```

Reply:
(It's not exactly the same, but at the same time non-objects won't have an `.id` either unless you try really hard for it)